app/stagemanager.js

```diff
@@ -125,22 +135,42 @@ class StageManager extends MyTrello {
     }
 
     orderLists(data) {
-        var position = 0,
-            self = this;
+      var deferred = Q.defer(),
+          position = 0,
+          ordering = [];
+
         data[0].forEach(function(stage, i) {
+          var stageDefer = Q.defer();
+          ordering.push(stageDefer.promise);
             var appropriateList = data[1].find(function(list){
               return list.name == stage.name;
             });
             if (appropriateList) {
                 var url = "1/lists/" + appropriateList.id + "/pos";
-                self.t.put(url, { value: position }, function(e, data) {
-                    if (e) throw e;
+                self.t.put(url, { value: position }, function(err, data) {
+                  if (err) stageDefer.reject(err);
+                  stageDefer.resolve();
                 });
                 position++;
             }
         });
+        promiseWaterfall(ordering).then(function() {
```

Comment:
It is supposed to do it sequentially as opposed to just gating when they all resolve.